src/nvim/api/buffer.c

```diff
@@ -757,6 +756,115 @@ void nvim_buf_set_text(uint64_t channel_id, Buffer buffer, Integer start_row, In
   try_end(err);
 }
 
+/// Gets a range from the buffer.
+///
+/// This differs from |nvim_buf_get_lines()| in that it allows retrieving only
+/// portions of a line.
+///
+/// Indexing is zero-based. Column indices are end-exclusive.
+///
+/// Prefer |nvim_buf_get_lines()| when retrieving entire lines.
+///
+/// @param channel_id
+/// @param buffer           Buffer handle, or 0 for current buffer
+/// @param start_row        First line index
+/// @param start_column     Starting byte offset of first line
+/// @param end_row          Last line index
+/// @param end_column       Ending byte offset of last line (exclusive)
+/// @param opts             Optional parameters. Currently unused.
+/// @param[out] err         Error details, if any
+/// @return Array of lines, or empty array for unloaded buffer.
+ArrayOf(String) nvim_buf_get_text(uint64_t channel_id, Buffer buffer,
+                                  Integer start_row, Integer start_col,
+                                  Integer end_row, Integer end_col,
+                                  Dictionary opts FUNC_ATTR_UNUSED,
```

Comment:
shouldn't be unused. rather explicitly check it is empty (as all keys are currently invalid).